code/game/objects/items/storage/belt.dm

```diff
@@ -635,31 +636,28 @@
 	SIGNAL_HANDLER
 	COOLDOWN_START(src, resheath_cooldown, 10 SECONDS)
 
-/datum/action/cooldown/spell/pointed/blade_counter
+/datum/action/innate/blade_counter
 	name = "Counterattack"
 	desc = "Anticipate an enemy's attack and strike back with your sheathed blade."
+	button_icon = 'icons/mob/actions/actions_spells.dmi'
 	button_icon_state = "declaration"
 	ranged_mousepointer = 'icons/effects/mouse_pointers/honorbound.dmi'
 
-	cooldown_time = 60 SECONDS
-	spell_requirements = NONE
-	antimagic_flags = NONE
-	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_LYING
+	enable_text = "You prepare to counterattack a target..."
+	disable_text = "You relax your stance."
 
-	active_msg = "You prepare to counterattack a target..."
-	deactive_msg = "You relax your stance."
-	cast_range = 2
+	click_action = TRUE
 
 	var/datum/weakref/blade_sheath_ref
 	var/datum/weakref/eyed_fool
 
-/datum/action/cooldown/spell/pointed/blade_counter/can_cast_spell(feedback = TRUE)
+/datum/action/innate/blade_counter/IsAvailable(feedback = TRUE)
 	if(!isliving(owner))
 		return FALSE
 	var/obj/item/storage/belt/sheath/owners_sheath = owner.get_item_by_slot(ITEM_SLOT_BELT)
```

Comment:
You can use ``item_action_slot_check`` on the sheath to control when the action is granted without needing to check it here